src/iidm/ThreeWindingsTransformerLeg.cpp

```diff
@@ -0,0 +1,215 @@
+    }
+    return *m_ratioTapChanger;
+}
+
+unsigned long Leg::getRegulatingTapChangerCount() const {
+    unsigned long count = 0;
+
+    const Leg& leg1 = m_transformer.get().getLeg1();
+    count += (leg1.hasPhaseTapChanger() && leg1.getPhaseTapChanger().isRegulating()) ? 1 : 0;
+    count += (leg1.hasRatioTapChanger() && leg1.getRatioTapChanger().isRegulating()) ? 1 : 0;
+
+    const Leg& leg2 = m_transformer.get().getLeg2();
+    count += (leg2.hasPhaseTapChanger() && leg2.getPhaseTapChanger().isRegulating()) ? 1 : 0;
+    count += (leg2.hasRatioTapChanger() && leg2.getRatioTapChanger().isRegulating()) ? 1 : 0;
+
+    const Leg& leg3 = m_transformer.get().getLeg3();
+    count += (leg3.hasPhaseTapChanger() && leg3.getPhaseTapChanger().isRegulating()) ? 1 : 0;
+    count += (leg3.hasRatioTapChanger() && leg3.getRatioTapChanger().isRegulating()) ? 1 : 0;
+
+    return count;
+}
+
+const Terminal& Leg::getTerminal() const {
+    return m_transformer.get().Connectable::getTerminal(m_legNumber - 1);
+}
+
+Terminal& Leg::getTerminal() {
+    return m_transformer.get().Connectable::getTerminal(m_legNumber - 1);
+}
+
+const std::string& Leg::getTypeDescription() const {
+    static std::array<std::string, 3> s_typeDescriptions {{
+                                                              u8"3 windings transformer leg1",
+                                                              u8"3 windings transformer leg2",
+                                                              u8"3 windings transformer leg3"
+                                                          }};
+
+    return s_typeDescriptions[m_legNumber - 1];
+}
+
+double Leg::getX() const {
+    return m_x;
+}
+
+bool Leg::hasPhaseTapChanger() const {
+    return static_cast<bool>(m_phaseTapChanger);
+}
+
+bool Leg::hasRatioTapChanger() const {
+    return static_cast<bool>(m_ratioTapChanger);
+}
+
+CurrentLimitsAdder<const std::nullptr_t, Leg> Leg::newCurrentLimits() {
+    return CurrentLimitsAdder<const std::nullptr_t, Leg>(nullptr, *this);
+}
+
+Leg& Leg::setB(double b) {
+    m_b = checkB(*this, b);
+    return *this;
+}
+
+void Leg::setCurrentLimits(std::nullptr_t /*side*/, std::unique_ptr<CurrentLimits> limits) {
+    m_limits = std::move(limits);
+}
+
+Leg& Leg::setG(double g) {
+    m_g = checkG(*this, g);
+    return *this;
+}
+
+Leg& Leg::setR(double r) {
+    m_r = checkR(*this, r);
+    return *this;
+}
+
+Leg& Leg::setRatedS(double ratedS) {
+    m_ratedS = checkRatedS(*this, ratedS);
+    return *this;
+}
+
+Leg& Leg::setRatedU(double ratedU) {
+    m_ratedU = checkRatedU(*this, ratedU);
+    return *this;
+}
+
+PhaseTapChangerAdder Leg::newPhaseTapChanger() {
+    return PhaseTapChangerAdder(*this);
+}
+
+RatioTapChangerAdder Leg::newRatioTapChanger() {
+    return RatioTapChangerAdder(*this);
+}
+
+Leg& Leg::setX(double x) {
+    m_x = checkX(*this, x);
+    return *this;
+}
+
+void Leg::setPhaseTapChanger(std::unique_ptr<PhaseTapChanger> phaseTapChanger) {
```

Comment:
Could you create an issue for the next release: we are not consistent in the way we pass unique_ptr (sometimes by rvalue). We need to decide what is our style

Reply:
https://github.com/powsybl/powsybl-iidm4cpp/issues/234